conformance/lang/expressions/range-expr/range_expr_exclusive.balt

```diff
@@ -0,0 +1,4979 @@
+
+    foreach int i in range {
+        count = count + 1;
+    }
+    io:println(count); // @output 15
+}
+
+Test-Case: output
+Description: Test shift expressions as expressions in range expression.
+Labels: range-expr, iterable-object, DecimalNumber, int, shift-expr, additive-expr, object-type, record-type, 
+        optional-type
+
+function init() {
+    int count = 0;
+
+    object {
+        *object:Iterable;
+        public isolated function iterator() returns object {
+            public isolated function next() returns (record {|int value;|}?);
+        };
+    } range = 1<<2 ..< 100>>4;
+
+    foreach int i in range {
+        count = count + 1;
+    }
+    io:println(count); // @output 2
+}
+
+Test-Case: output
+Description: Test conditional expressions as expressions in range expression.
+Labels: range-expr, iterable-object, conditional-expr, DecimalNumber, error, int, additive-expr, nil-literal,
+        optional-type, union-type, object-type, record-type
+
+function init() {
+    int|error a = 10;
+    int? b = ();
+    int count = 0;
+
+    object {
+        *object:Iterable;
+        public isolated function iterator() returns object {
+            public isolated function next() returns (record {|int value;|}?);
+        };
+    } range = (a is error ? 5 : a) ..< (b ?: 25);
+
+    foreach int i in range {
+        count = count + 1;
+    }
+    io:println(count); // @output 15
+}
+
+Test-Case: output
+Description: Test lang.int constants as expressions in range expression.
+Labels: range-expr, iterable-object, int, DecimalNumber, int:SIGNED8_MAX_VALUE, int:UNSIGNED8_MAX_VALUE, additive-expr, 
+        object-type, record-type, optional-type
+
+function init() {
+    int count = 0;
+
+    object {
+        *object:Iterable;
+        public isolated function iterator() returns object {
+            public isolated function next() returns (record {|int value;|}?);
+        };
+    } range = int:SIGNED8_MAX_VALUE ..< int:UNSIGNED8_MAX_VALUE;
+
+    foreach int i in range {
+        count = count + 1;
+    }
+    io:println(count); // @output 128
+}
+
+Test-Case: output
+Description: Test isolatedness of range expression.
+Fail-Issue: ballerina-platform/ballerina-lang#33163
+Labels: range-expr, iterable-object, object-type, additive-expr, int, DecimalNumber, function-call-expr, record-type,
+        optional-type, lock-stmt
+
+isolated int i = 1;
+
+function init() {
+    int count = 0;
+    foreach int i in getRange() {
+        count = count + 1;
+    }
+    io:println(count); // @output 10
+}
+
+function getRange() returns object {
+    *object:Iterable;
+    public isolated function iterator() returns object {
+        public isolated function next() returns (record {|int value;|}?);
+    };
+} {
+    lock {
+        return i ..< i + 9;
+    }
+}
+
+Test-Case: output
+Description: Test isolatedness of range expression.
+Fail-Issue: ballerina-platform/ballerina-lang#33163
```

Comment:
Shall we update the tests to test to test range-expr being an isolated expression as mentioned in https://github.com/ballerina-platform/ballerina-lang/issues/33163#issuecomment-940742860.

We need the current tests too, but they don't quite test "A range-expr is always isolated.".

Reply:
Have already been added. 
range_expr_exclusive line 4892
range_expr_inclusive line 4887

Reply:
Shall we update the descriptions of these tests then? To say we're testing the isolatedness of the object and the method.